transport/sctp/SctpAssociationImpl.cpp

```diff
@@ -371,7 +371,11 @@ SctpAssociationImpl::SctpAssociationImpl(size_t logId,
         _local.inboundStreamCount = cookie.inboundStreams;
         _local.outboundStreamCount = cookie.outboundStreams;
     }
+}
 
+// must only be called once and immediately after creation due to COOKIE ECHO received
+void SctpAssociationImpl::onCookieEcho(const SctpPacket& cookieEcho, const uint64_t timestamp)
```

Comment:
This was the main problem. Processing the echo caused callbacks onto IEvents before the object had been returned and still in constructor.